src/pilot/assign_role.py

```diff
@@ -1023,9 +1028,14 @@ def generate_osd_config_without_journals(controllers, drives):
         'osd_objectstore': 'bluestore',
         'devices': []}
     for drive in drives:
-        drive_device_name = get_by_path_device_name(
-             drive, controllers)
-        osd_config['devices'].append(drive_device_name)
+        # Get by-path device name for NVMe drives
+        if drive.controller.startswith('PCIe'):
```

Comment:
I think we should always use the same method to determine if a drive is an NVMe drive in all cases.  Here's we check the controller, but above, we check the device_protocol.  I think the better way to do it is to check the device_protocol.  Can you make a new method called is_nvme_drive() and perform the check in there?  Then you can call it from both places, and we can be sure that NVMe drives are detected the same way everywhere.

Reply:
Hey Chris, 
Check for NVMe drives from get_drives() will be done through device_protocol as it is done for only physical disks while checking NVMe drives from generate_osd_config_without_journals() cannot be done on similar grounds as the drives would be sum of physical disks and virtual disks where virtual disks have no attribute named "device_protocol". Therefore in this particular function drive.controller is checked instead of drive.device_protocol.

However , I tried implementing your review comment and tested positively but not sure whether it will be all-time valid condition. It is -

def is_nvme_drive(disk):
    return True if (not disk.id.startswith("Disk.Virtual") and disk.device_protocol and disk.device_protocol.startswith("NVMe") else False

So when is_nvme_drive() is called from generate_osd_config_without_journals() , it will only extract PCIeSSD disks from a list of physical disks and virtual disks.